File: backend/src/test/java/lan/dk/podcastserver/service/properties/ApiTest.java
package lan.dk.podcastserver.service.properties;

import org.junit.Before;
import org.junit.Test;

import static lan.dk.podcastserver.service.properties.ApiAssert.assertThat;

/**
 * Created by kevin on 13/04/2016 for Podcast Server
 */
public class ApiTest {

    Api api;

    @Before
    public void beforeEach() {
        api = new Api();
    }

    @Test
    public void should_have_default_value() {
        /* Given */
        /* When */
        /* Then */
        assertThat(api)
                .hasYoutube(null)
                .hasDailymotion(null);
    }

    @Test
    public void should_have_specified_values() {
        /* Given */
        String youtubeKey = "YoutubeKey";
        String dailymotionKey = "dailymotionKey";

        /* When */
        api.setDailymotion(dailymotionKey).setYoutube(youtubeKey);

        /* Then */
        assertThat(api)
                .hasYoutube(youtubeKey)
                .hasDailymotion(dailymotionKey);
    }

}

File: backend/src/test/kotlin/com/github/davinkevin/podcastserver/service/properties/ApiTest.kt
package com.github.davinkevin.podcastserver.service.properties

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

/**
 * Created by kevin on 13/04/2016 for Podcast Server
 */
class ApiTest {

    @Test
    fun should_have_default_value() {
        /* Given */
        /* When */
        val api = Api()
        /* Then */

        assertThat(api.youtube).isEqualTo("")
    }

    @Test
    fun should_have_specified_values() {
        /* Given */
        val youtubeKey = "YoutubeKey"

        /* When */
        val api = Api(
                youtube = youtubeKey
        )

        /* Then */
        assertThat(api.youtube).isEqualTo(youtubeKey)
    }

}
